### PR TITLE
remove WM5 from headers

### PR DESCRIPTION
**Pull request by timosachsenberg**
# Description

To prevent symbols being exported.
RegressionUtils.h was removed for same reasons (did not safe code anyways).
As well as the test which was not run anyways.

# Checklist:
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.
- Or click on the action: e.g., for clang-format linting

# Note:
- Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

# Advanced commands (admins / reviewer only):
- /rebase will try to rebase the PR on the current develop branch.
- /reformat (experimental) applies the clang-format style changes as additional commit
- setting the label NoJenkins will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)

### CODE REVIEW

File: src/openms/source/MATH/STATISTICS/QuadraticRegression.cpp [hunk 1]
```diff
@@ -34,21 +34,120 @@
...
+      double weight = *wIter;
+
+      sumX += weight * x;
+      sumXX += weight * x * x;
+      sumXXX += weight * x * x * x;
+      sumXXXX += weight * x * x * x * x;
+
+      sumY += weight * y;
+      sumXY += weight * x * y;
+      sumXXY += weight * x * x * y;
```
Comment (jpfeuffer) [inline src/openms/source/MATH/STATISTICS/QuadraticRegression.cpp:121]: this also screams for optimization, unless the compiler is very smart

**Review by jpfeuffer — CHANGES_REQUESTED** [review]

File: src/openms/source/MATH/STATISTICS/QuadraticRegression.cpp (hunk 1 above)
Comment (jpfeuffer) [inline src/openms/source/MATH/STATISTICS/QuadraticRegression.cpp:121]:
```suggestion
      summand = weight;
      sumY += summand * y;
      summand *= x;
      sumX += summand;
      sumXY += summand * y;
      summand *= x;
      sumXX += summand;
      sumXXY += summand * y;
      summand *= x;
      sumXXX += summand;
      summand *= x;
      sumXXXX += summand;
```

Comment (jpfeuffer) [reply src/openms/source/MATH/STATISTICS/QuadraticRegression.cpp:121]: but I don't know if worth it.

Comment (timosachsenberg) [reply src/openms/source/MATH/STATISTICS/QuadraticRegression.cpp:121]: My guess is that the compiler is very smart here.

**Comment by jpfeuffer** [comment]
https://stackoverflow.com/questions/60463014/cmath-error-on-macos-c17-call-to-abs-ambiguous
please leave fabs

**Comment by timosachsenberg** [comment]
> https://stackoverflow.com/questions/60463014/cmath-error-on-macos-c17-call-to-abs-ambiguous please leave fabs

I thought this only holds for integral types and not for double. Are you sure this causes an error? Or is cmath missing?

**Comment by jpfeuffer** [comment]
Yes it's in Jenkins.
Abs only works for floats since c17 and apparently not in all versions of Apples libc++

File: src/openms/source/FILTERING/SMOOTHING/LowessSmoothing.cpp [hunk 2]
```diff
@@ -116,7 +117,7 @@ namespace OpenMS
     }
 
     // 0 <= u < t; u is regarded as 0.0 if fabs(u) falls below epsilon
-    if ((fabs(u) < std::numeric_limits<double>::epsilon() || (0.0 < u)) && (u < t))
+    if ((std::abs(u) < std::numeric_limits<double>::epsilon() || (0.0 < u)) && (u < t))
```
Comment (timosachsenberg) [inline src/openms/source/FILTERING/SMOOTHING/LowessSmoothing.cpp:120]:
```suggestion
    if ((fabs(u) < std::numeric_limits<double>::epsilon() || (0.0 < u)) && (u < t))
```

File: src/openms/source/FILTERING/SMOOTHING/LowessSmoothing.cpp [hunk 3]
```diff
@@ -79,7 +80,7 @@ namespace OpenMS
       // Size inner_idx = 0;
       for (Size inner_idx = 0; inner_idx < input_size; ++inner_idx)
       {
-        distances[inner_idx] = std::fabs(input_x[outer_idx] - input_x[inner_idx]);
+        distances[inner_idx] = std::abs(input_x[outer_idx] - input_x[inner_idx]);
```
Comment (timosachsenberg) [inline src/openms/source/FILTERING/SMOOTHING/LowessSmoothing.cpp:83]:
```suggestion
        distances[inner_idx] = fabs(input_x[outer_idx] - input_x[inner_idx]);
```

**Comment by jpfeuffer** [comment]
I think it has to be std::fabs

File: src/openms/source/MATH/STATISTICS/QuadraticRegression.cpp (hunk 1 above)
Comment (cbielow) [reply src/openms/source/MATH/STATISTICS/QuadraticRegression.cpp:120]: https://godbolt.org/z/9vnah9Gb9

it is (as expected), but only in `-O2`. So in debug mode, we're stuck with slow code.

**Comment by jpfeuffer** [comment]
Should be fine. Windows out of space.

**Pull request merged**